### PR TITLE
Eigenvector centrality

### PR DESCRIPTION
**Pull request by vigna**
This pull request implements eigenvector centrality. As discussed in https://github.com/jgrapht/jgrapht/issues/1013, AlphaCentrality is not really computing eigenvector centrality. This class implements the power method with 𝓁₂ normalization (the most commonly used normalization).

Tests have been created using the Maple mathematical software package to compute eigenvector centrality on sample matrices.

I think that it would be a good idea to have a KatzCentrality class doing essentially what AlphaCentrality does, but pointing at the right papers and dispensing with the exogenous factor as a scalar. In Katz centrality, any constant vector will give the same centrality, just scaled by the constant contained in the vector. Having a scalar exogenous factor will make the user believe that by changing it the centrality computed will somehow change, but all that will happen is a rescaling. 

I refrained from touching the documentation in AlphaCentrality, but probably it would also be a good idea to eliminate the part related to eigenvector centrality, as the results would be completely different from those returned by this class.

### CODE REVIEW

**Comment by d-michail** [comment]
I agree. The sentence about eigenvector centrality should be removed. Regarding the Katz centrality, due to deprecation policies, we need to keep this class at least for another release. 

Actions:
  - Remove the sentence about eigenvector centrality from `AlphaCentrality`
  - Deprecate `AlphaCentrality` (add appropriate annotation in class and javadoc)
  - Copy AlphaCentrality to KatzCentrality and remove the constructor with the scalar value exogenous factor.
  - Fix the docs in KatzCentrality to point to the right papers.
  - Tests can simply be renamed and map to the new implementations.

Would it be possible to perform at least the first step in #1015 ? This would make that mergeable.

**Comment by vigna** [comment]
I can do everything if you like.

**Comment by vigna** [comment]
OK, that should be it.

File: jgrapht-core/src/main/java/org/jgrapht/alg/scoring/EigenvectorCentrality.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/*
+ * (C) Copyright 2020, by Sebastiano Vigna.
```
Comment (d-michail) [inline jgrapht-core/src/main/java/org/jgrapht/alg/scoring/EigenvectorCentrality.java:2]: We use the following format across the library

(C) Copyright startyear-endyear, by MainAuthorName and Contributors.

File: jgrapht-core/src/test/java/org/jgrapht/alg/scoring/EigenvectorCentralityTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
+/*
+ * (C) Copyright 2020, by Sebastiano Vigna.
```
Comment (d-michail) [inline jgrapht-core/src/test/java/org/jgrapht/alg/scoring/EigenvectorCentralityTest.java:2]: Same comment

**Comment by d-michail** [comment]
Thanks!

**Pull request merged**